filament/src/View.cpp

```diff
@@ -800,6 +802,19 @@ void FView::renderShadowMaps(FEngine& engine, FEngine::DriverApi& driver, Render
     mShadowMapManager.render(engine, *this, driver, pass);
 }
 
+void FView::commitFrameHistory(FEngine& engine) noexcept {
+    // Here we need to destroy resources in mFrameHistory.back()
+    // and then push the new history entry to the history stack
+    mFrameHistory.commit();
+}
+
+void FView::drainFrameHistory(FEngine& engine) noexcept {
+    // make sure we free all resources in the history
```

Comment:
What types of resources are you planning on storing in `FrameHistoryEntry`?

Reply:
It's going to be things like `Handle<>` which need to be destroyed on the main thread and need a `DriverApi&` for the destruction.